pkg/config/derive_test.go

```diff
@@ -0,0 +1,190 @@
+package config
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+)
+
+func TestDerivedCloudProviderConf(t *testing.T) {
+	tests := []struct {
+		cs   api.OpenShiftManagedCluster
+		name string
+		want []byte
+	}{
+		{
+			name: "one",
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{AzProfile: &api.AzProfile{
+					TenantID:       "tenant",
+					SubscriptionID: "sub",
+					ResourceGroup:  "rg",
+				}, ServicePrincipalProfile: &api.ServicePrincipalProfile{
+					ClientID: "client_id",
+					Secret:   "client_secrett",
+				}},
+				Location: "eastus",
+			},
+			want: []byte(`aadClientId: client_id
+aadClientSecret: client_secrett
+location: eastus
+primaryScaleSetName: ss-compute
+resourceGroup: rg
+securityGroupName: nsg-compute
+subscriptionId: sub
+tenantId: tenant
+vmType: vmss
+`),
+		},
+	}
+
+	for _, tt := range tests {
+		got, err := Derived.CloudProviderConf(&tt.cs)
+		if err != nil {
+			t.Error(err)
+			return
+		}
+		if !reflect.DeepEqual(got, tt.want) {
+			t.Errorf("derived.CloudProviderConf() = \"%v\", want \"%v\"", string(got), string(tt.want))
+		}
+	}
+}
+
+func TestDerivedMasterLBCNamePrefix(t *testing.T) {
+	cs := api.OpenShiftManagedCluster{
+		Properties: &api.Properties{FQDN: "bar.baz"},
+	}
+	if got := Derived.MasterLBCNamePrefix(&cs); got != "bar" {
+		t.Errorf("derived.MasterLBCNamePrefix() = %v, want %v", got, "bar")
+	}
+}
+
+func TestDerivedRouterLBCNamePrefix(t *testing.T) {
+	tests := []struct {
+		cs   api.OpenShiftManagedCluster
+		want string
+	}{
+		{
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{
+					RouterProfiles: []api.RouterProfile{
+						{
+							FQDN: "one.two.three",
+						},
+						{
+							FQDN: "not.this",
```

Comment:
(we currently validate against multiple RouterProfiles)

Reply:
ok, i have removed the second entry now